src/Features/Core/Portable/SimplifyInterpolation/AbstractSimplifyInterpolationCodeFixProvider.cs

```diff
@@ -0,0 +1,110 @@
+        protected abstract TInterpolationSyntax WithFormatClause(TInterpolationSyntax interpolation, TInterpolationFormatClause? formatClause);
+        protected abstract string Escape(TInterpolatedStringExpressionSyntax interpolatedString, string formatString);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.RequireSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+            foreach (var diagnostic in diagnostics)
+            {
+                var loc = diagnostic.AdditionalLocations[0];
+                var interpolation = semanticModel.GetOperation(loc.FindNode(getInnermostNodeForTie: true, cancellationToken)) as IInterpolationOperation;
+                if (interpolation?.Syntax is TInterpolationSyntax interpolationSyntax &&
+                    interpolationSyntax.Parent is TInterpolatedStringExpressionSyntax interpolatedString)
+                {
+                    Helpers.UnwrapInterpolation<TInterpolationSyntax, TExpressionSyntax>(
+                        document.GetRequiredLanguageService<IVirtualCharService>(), interpolation, out var unwrapped,
+                        out var alignment, out var negate, out var formatString, out _);
+
+                    if (unwrapped == null)
+                        continue;
+
+                    alignment = negate ? (TExpressionSyntax)generator.NegateExpression(alignment) : alignment;
+
+                    editor.ReplaceNode(
+                        interpolationSyntax,
+                        Update(generator, interpolatedString, interpolationSyntax, unwrapped, alignment, formatString));
+                }
+            }
+        }
+
+        private TInterpolationSyntax Update(
+            SyntaxGenerator generator, TInterpolatedStringExpressionSyntax interpolatedString,
+            TInterpolationSyntax interpolation, TExpressionSyntax unwrapped,
+            TExpressionSyntax? alignment, string? formatString)
+        {
+            var result = WithExpression(interpolation, unwrapped);
+            if (alignment != null)
+            {
+                result = WithAlignmentClause(
+                    result,
+                    (TInterpolationAlignmentClause)generator.InterpolationAlignmentClause(alignment));
+            }
+
+            if (formatString != null)
+            {
+                result = WithFormatClause(result, formatString == ""
+                    ? null
```

Comment:
will a format clause of `""` ever produce something different? I don't see why we keep it

Reply:
not certain what ths means.  :)

Reply:
`WithFormatClause(null, ...)` just seems weird to me from reading it. Maybe I misunderstood what it does?

Reply:
Good catch. Investigating.

Reply:
removed.